aggregator_core/src/datastore/test_util.rs

```diff
@@ -130,6 +134,35 @@ impl EphemeralDatastore {
             LessSafeKey::new(UnboundKey::new(&AES_128_GCM, &self.datastore_key_bytes).unwrap());
         Crypter::new(Vec::from([datastore_key]))
     }
+
+    pub async fn downgrade(&self, target: i64) {
+        let mut connection = PgConnection::connect(&self.connection_string)
+            .await
+            .unwrap();
+
+        let current_version = connection
+            .list_applied_migrations()
+            .await
+            .unwrap()
+            .iter()
+            .max_by(|a, b| a.version.cmp(&b.version))
+            .unwrap()
+            .version;
+        if target >= current_version {
+            panic!(
+                "target version ({}) must be less than the current database version ({})",
+                target, current_version,
+            );
+        }
+
+        // Run down migrations one at a time to provide better context when
+        // one fails.
+        for v in (target..current_version).rev() {
+            if let Err(e) = self.migrator.undo(&mut connection, v).await {
+                panic!("failed to downgrade to version {}: {}", v, e);
```

Comment:
Unsure if there's any preference for `.expect(format!("").as_str())` or just a direct `panic!()` here.

Reply:
Using the panic macro is fine by me

Reply:
Ah, of the options `unwrap_or_else` looks cleanest to me, avoids the `if let` construct (pardon the Rust noobness).